src/Processors/Formats/IOutputFormat.h

```diff
@@ -85,6 +95,55 @@ class IOutputFormat : public IProcessor
         }
     }
 
+    void writeSuffixIfNot()
+    {
+        if (need_write_suffix)
+        {
+            writeSuffix();
+            need_write_suffix = false;
+        }
+    }
+
+    /// Methods-helpers for parallel formatting.
+
+    /// Set the real number of first row in the first chunk of data.
```

Comment:
But it is not really for the first chunk of data, but for every chunk of data?

Reply:
Not exacetely, the first row number in the second chunk is just the first row number in the first chunk + the number of rows in the first chunk, and etc. Maybe just bad comment or field/method name (will think about better naming)

Reply:
I see